### PR TITLE
Add layouts guide

### PR DESCRIPTION
**Pull request by jxnblk**
- Pulls the layouts section of the variants guide into its own page
- Also sets a global layout for the docs site

[Rendered version](https://deploy-preview-247--theme-ui.netlify.com/guides/layouts)

### CODE REVIEW

File: packages/docs/src/pages/guides/layouts.mdx [hunk 1]
```diff
@@ -0,0 +1,130 @@
+export const title = 'Layouts'
+
+# Layouts
+
+By leveraging the [variants](/guides/variants) feature, you can create custom themeable layout components.
+The components can include some default styles,
+but still allow users to override styles such as colors or typography, by changing values in the theme object.
+
+To start, create a layout component that will pages in your application.
```
Comment (johnletey) [inline packages/docs/src/pages/guides/layouts.mdx]: @jxnblk This doesn't sound quite right

File: packages/docs/src/pages/guides/layouts.mdx [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+Feel free to customize the content and default styles however you need.
```
Comment (johnletey) [inline packages/docs/src/pages/guides/layouts.mdx]:
```suggestion
Feel free to customize the content and default styles however you like.
```

File: packages/docs/src/pages/guides/layouts.mdx [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+        width: '100%',
+        variant: 'layout.footer',
+      }}>
+      Footer content
+    </footer>
+  </div>
+```
+
+The `variant` properties above will look for styles in `theme.layout`.
+When no styles are defined, the default styles will still be used.
```
Comment (johnletey) [inline packages/docs/src/pages/guides/layouts.mdx]:
```suggestion
When no styles are defined, the default styles will be used.
```

File: packages/docs/src/pages/guides/layouts.mdx [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+
+If a user wants to customize the layout styles, they can add overrides to their theme object.
+
+For example, the following will add colors to the header and footer.
```
Comment (johnletey) [inline packages/docs/src/pages/guides/layouts.mdx]:
```suggestion
For example, the following will add custom colors to the header and footer.
```

File: packages/docs/src/pages/guides/layouts.mdx (hunk 1 above)
Comment (jxnblk) [reply packages/docs/src/pages/guides/layouts.mdx]: Missed a word
```suggestion
To start, create a layout component that will wrap pages in your application.
```

Comment (johnletey) [reply packages/docs/src/pages/guides/layouts.mdx]: Yup ... sounds a lot better!

File: packages/docs/src/pages/guides/layouts.mdx (hunk 3 above)
Comment (johnletey) [reply packages/docs/src/pages/guides/layouts.mdx]: @jxnblk And this one?

**Pull request merged**